### PR TITLE
Enabling encryption: Handle external account decryption in PermsStore, add docs for reversing OOB migrations.

### PR DESCRIPTION
**Pull request by arussellsaw**
probably should be two PRs but it's small and covering issues related to enabling encryption, i was getting an error from repo-updater after configuring encryption keys, looks like we just needed to pass keys to the PermsStore

### CODE REVIEW

File: enterprise/internal/database/perms_store.go [hunk 1]
```diff
@@ -1115,7 +1118,8 @@ SELECT
     id, user_id,
     service_type, service_id, client_id, account_id,
     auth_data, account_data,
-    created_at, updated_at
+    created_at, updated_at,
+	encryption_key_id
```
Comment (ryanslade) [inline enterprise/internal/database/perms_store.go:1122]: nit

```suggestion
	  encryption_key_id
```

File: internal/database/external_accounts.go [hunk 2]
```diff
@@ -491,11 +491,16 @@ func MaybeDecrypt(ctx context.Context, key encryption.Key, data, keyIdent string
...
 	}
+	if data == "" {
+		return data, nil
+	}
 	if key == nil {
 		return data, fmt.Errorf("couldn't decrypt encrypted data, key is nil")
 	}
 	decrypted, err := key.Decrypt(ctx, []byte(data))
 	if err != nil {
+		fmt.Println(data)
```
Comment (ryanslade) [inline internal/database/external_accounts.go:502]: Debugging code?

**Review by ryanslade — APPROVED** [review]

File: enterprise/internal/database/perms_store.go [hunk 3]
```diff
@@ -6,6 +6,9 @@ import (
 	"encoding/json"
 	"time"
 
+	"github.com/sourcegraph/sourcegraph/internal/database"
```
Comment (unknwon) [inline enterprise/internal/database/perms_store.go:9]: I'm paranoid about import grouping... these two imports should move down :D

File: enterprise/internal/database/perms_store.go [hunk 4]
```diff
@@ -1115,7 +1118,8 @@ SELECT
     id, user_id,
     service_type, service_id, client_id, account_id,
     auth_data, account_data,
-    created_at, updated_at
+    created_at, updated_at,
+	  encryption_key_id
```
Comment (unknwon) [inline enterprise/internal/database/perms_store.go:1122]: Mixed use of tabs and spaces

CleanShot 2021-04-08 at 00 42 04@2x

**Pull request merged**